### PR TITLE
Adding types for existing node module "express-list-endpoints"

### PR DESCRIPTION
**Pull request by sjoseph7**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`

### CODE REVIEW

**Comment by sjoseph7** [comment]
I've re-written my code to more closely follow the code style in the "express-less" files.

**Comment by sjoseph7** [comment]
DefinitelyTyped.BenchmarkPR seems to have stalled; resetting pipeline.

**Comment by sjoseph7** [comment]
DefinitelyTyped.BenchmarkPR seems to have stalled; resetting pipeline.

File: types/express-list-endpoints/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// Type definitions for express-list-endpoints 4.0
+// Project: https://github.com/AlbertoFdzM/express-list-endpoints
+// Definitions by: S Joseph <https://github.com/sjoseph7>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import express = require('express');
+
+export function getEndpoints(app: express.Express): Endpoint[];
```
Comment (amcasey) [inline types/express-list-endpoints/index.d.ts]: To match the underlying library, it looks like you want
```ts
function getEndpoints //...
export = getEndpoints
```

**Review by amcasey — CHANGES_REQUESTED** [review]

**Review by sjoseph7 — COMMENTED** [review]
I've updated the code to represent the commit that I _thought_ I had...

File: types/express-list-endpoints/tslint.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "adjacent-overload-signatures": false,
```
Comment (amcasey) [inline types/express-list-endpoints/tslint.json]: Why disable all these lint rules?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/express-list-endpoints/tslint.json (hunk 2 above)
Comment (sjoseph7) [reply types/express-list-endpoints/tslint.json]: @amcasey I copied the rules from a neighboring and very similar types package when I ran into an issue while running `npm run lint express-list-endpoints`; after making some changes to the files and reviewing again, I'm left with just an "npm-naming" issue:

```
Error: <...>/DefinitelyTyped/types/express-list-endpoints/index.d.ts:1:1
ERROR: 1:1  npm-naming  d.ts file must have a matching npm package.
To resolve this error, either:
1. Change the name to match an npm package.
```

The necessary header is included and the express-list-endpoints module [exists on npm](https://www.npmjs.com/package/express-list-endpoints)... is there something I'm overlooking?

File: types/express-list-endpoints/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// Type definitions for express-list-endpoints 4.0
+// Project: express-list-endpoints
```
Comment (amcasey) [inline types/express-list-endpoints/index.d.ts]: I think you can fix the last lint issue by changing this to `Project: https://github.com/AlbertoFdzM/express-list-endpoints`

**Review by sjoseph7 — COMMENTED** [review]
Test "Project: https://github.com/AlbertoFdzM/express-list-endpoints" replacement

File: types/express-list-endpoints/index.d.ts (hunk 3 above)
Comment (sjoseph7) [reply types/express-list-endpoints/index.d.ts]: I'll give it a try.

**Review by sjoseph7 — COMMENTED** [review]
Test "Project: https://github.com/AlbertoFdzM/express-list-endpoints" replacement

File: types/express-list-endpoints/index.d.ts (hunk 3 above)
Comment (sjoseph7) [reply types/express-list-endpoints/index.d.ts]: 👍

Comment (sjoseph7) [reply types/express-list-endpoints/index.d.ts]: @amcasey looks like it worked! Thanks for the suggestion.

**Review by amcasey — APPROVED** [review]

**Pull request merged**